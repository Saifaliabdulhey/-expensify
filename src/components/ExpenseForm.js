import React, { Component } from 'react'
import 'react-dates/initialize';
import moment from 'moment'
import {  SingleDatePicker } from 'react-dates';


export class ExpenseForm extends Component {

    constructor(props) {
        super(props);

        this.state = {
            description: props.expense ? props.expense.description :'',
            note: props.expense ? props.expense.note :'',
            amount: props.expense ? (props.expense.amount / 100).toString() :'',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: true,
            error: '',
        }
    }


    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({description}));
    }

    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({note}));
    }

    onAmountChange = (e) => {
        const amount = e.target.value;
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
         this.setState(() => ({amount})); 
        }
    }

    onDateChange = (createdAt) => {
        if(createdAt) {
            this.setState(() => ({createdAt}));
        }
    }

    onFocusChange = ({focused}) => {
        this.setState(() => ({calanderFocused: focused}));
    }

    onSubmit = (e) => {
        e.preventDefault();
        if(!this.state.description || !this.state.amount) {
            this.setState(() => ({error: 'Please provide description and amount'}));
        }else {
            this.setState(() => ({error: ''}));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            })
        }
    }
  render() {
    return (
          
          <form className="form" onSubmit={this.onSubmit}>
          {this.state.error && <p className="form__error">{this.state.error}</p>}
              <input
              type="text"
              placeholder="Description"
              className="text-input"
              autoFocus
              value= {this.state.description}
              onChange={this.onDescriptionChange}
              />

              <input 
              type="text"
              className="text-input"
              placeholder="Amount"
              value={this.state.amount}
              onChange={this.onAmountChange}
              />

              <textarea
              className="textarea"
              placeholder="Add a note for your expense (Optional)"
              value={this.state.note}
              onChange={this.onNoteChange}
              >    
              </textarea>

           <div className="extra-changes">
              <SingleDatePicker
              date={this.state.createdAt}
              onDateChange={this.onDateChange}
              focused={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              numberOfMonths={1}
              isOutsideRange={()=> false}
              />

             <button className="button btn">Save Expense</button>
             </div>
          </form>
    )
  }
}

export default ExpenseForm