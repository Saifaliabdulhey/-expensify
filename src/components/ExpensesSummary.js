import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses.total';

export const ExpensesSummary = ({expenseCount, expensesTotal, wholeExpenses}) => {
    const expenseWord = expenseCount === 1 ? 'expense': 'expenses';
    const formattedExpenseTotal = numeral(expensesTotal / 100).format('$0,0,00');

    return (
        <div className="page-header">
        <div className="content-container">
        <h1 className="page-header__title">
        Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpenseTotal}</span></h1>
        <p>you have {expenseCount} visible {expenseWord} of <span>{wholeExpenses}</span> </p>
        <div className="page-header_actions">
        <Link className="button" to="/contact">Add Expense</Link>
        </div>
        </div>
        </div>
    );
};


const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        wholeExpenses: state.expenses.length
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
