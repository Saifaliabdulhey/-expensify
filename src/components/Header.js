import React  from 'react'
import {Link} from 'react-router-dom'
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({startLogout}) => (
         <header className="header">
            <div className="content-container" >
                <div className="header__content">
                    <Link className="header_title" to="/home" >
                        <h1>Expensify</h1>
                     </Link> 
                  <button className="button button--link"onClick={startLogout}>Logout</button> 
               </div>
            </div>
          </header>
    )

    const mapDispatchToProps = (dispatch) => ({
        startLogout:() => dispatch(startLogout())
    })

export default connect(undefined, mapDispatchToProps)(Header);
// <NavLink to="/portfolio" activeClassName='is-active' exact>Portfolio</NavLink>
// <NavLink to="/contact" activeClassName='is-active'>Create Expense</NavLink>