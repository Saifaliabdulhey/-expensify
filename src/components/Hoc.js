// import React from 'react'

// function Hoc(props) {
//   return (
//     <div>
//         <h1>Info</h1>
//         <p>The info is:{props.info} </p>
//     </div>
//   )
// }

// export default Hoc