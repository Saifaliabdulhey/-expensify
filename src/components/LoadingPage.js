import React from 'react'
import loader from '../images/loader.gif'

function LoadingPage() {
  return (
    <div className="loader">
    <img className="loader__image" src={loader} />
    </div>
  )
}

export default LoadingPage