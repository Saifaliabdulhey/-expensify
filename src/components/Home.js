import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";

function Home() {
  return (
    <div>
    <ExpensesSummary />
      <ExpenseListFilters />
      <ExpenseList />
    </div>
  );
}

export default Home;
