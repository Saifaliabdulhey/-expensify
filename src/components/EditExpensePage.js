import React from "react";
import { connect } from "react-redux";
import { setState } from "react"
import ExpenseForm from "./ExpenseForm";
import { startEditExpenses, startRemoveExpenses } from "../actions/expenses";
import Header from "./Header";
import Modal from 'react-modal';

export class EditExpensePage extends React.Component {

  constructor(props) {
    super(props);

    this.state = ({modal: true});

    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  onSubmit = (expense) => {
    this.props.startEditExpenses(this.props.expense.id, expense);
    this.props.history.push('/');
  }

  onRemove = () => {
    this.props.startRemoveExpenses({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  // Set up modal style for removeing expense

   openModal() {
     this.setState({ modal: false });
  }



   closeModal() {
    this.setState({ modal: true });
  }


  render() {
    return (
      <div className="genral">
      <Header />
      <div className="page-header">
        <div className="content-container">
          <h1>Edit Expense</h1>
        </div>
      </div>

      <div className="content-container">
      <ExpenseForm
        expense={this.props.expense}
        onSubmit = {this.onSubmit}
      />
      </div>
      <div className="content-container">
      <button className="button btn-c"onClick={this.openModal}>Remove Expense</button>
      </div>
      <Modal
      appElement={document.getElementById('root')}
      isOpen={!this.state.modal}
      onRequestClose={this.closeModal}
      closeTimeoutMS={200}
      className="modal"
      >
      <h3>You sure you want to delete this?</h3>
      <button className="modal__btn" onClick={this.onRemove}>yes</button>
      <button className="modal__btn" onClick={this.closeModal}>no</button>
      </Modal>
    </div>
    );
  }
};


const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      (expense) => expense.id === props.match.params.id
    ),
    
  };
  
};
const mapDispatchToProps = (dispatch, props) => ({
  startEditExpenses: (id, expense) => dispatch(startEditExpenses(id, expense)),
  startRemoveExpenses: (data) => dispatch(startRemoveExpenses(data))
})


export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
