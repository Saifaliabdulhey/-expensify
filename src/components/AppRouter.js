import React from 'react'
import Home from './Home';
import createHistory from 'history/createBrowserHistory';
// import Portfolio from './Portfolio';
import EditExpensePage from './EditExpensePage';
import NotFoundPage from './NotFoundPage';
import ReduxExpensify from './ReduxExpensify'
import { Router, Route, Link, Switch, NavLink } from 'react-router-dom';
import Contact  from './Contact';
import LoginPage from './LoginPage'
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';


export const history = createHistory();

function AppRouter() {
  return (
 <Router history={history}>
      <Switch>
      <PublicRoute path="/" exact component={LoginPage} />
      <PrivateRoute path="/home" exact component={Home} />
      <PrivateRoute path="/redux" exact component={ReduxExpensify} />
      <PrivateRoute path="/contact" component={Contact} />
      <Route path="/edit/:id" component={EditExpensePage} />
      <Route  component={NotFoundPage} />
      </Switch>
  
    </Router>
    
  )
}
   // <Route path="/portfolio" exact component={Portfolio} />
  export default AppRouter;