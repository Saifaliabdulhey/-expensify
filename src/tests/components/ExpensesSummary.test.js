import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render ExpensesSummary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={237667675}/>);
    expect(wrapper).toMatchSnapshot();
})