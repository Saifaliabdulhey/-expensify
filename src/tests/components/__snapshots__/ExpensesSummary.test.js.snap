// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with multiple expense 1`] = `ShallowWrapper {}`;

exports[`should correctly render ExpensesSummary with one expense 1`] = `ShallowWrapper {}`;
